File: Backend-Spring/src/main/java/com/phase2/javaProject_Phase2/beans/Company.java
package com.phase2.javaProject_Phase2.beans;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.validator.constraints.Length;

import java.util.List;

@Entity
@Table(name = "COMPANIES")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Company {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false, length = 20)
    @Length(min = 2, max = 40)
    private String name;

    @Column(unique = true, nullable = false)
    private String email;

    @Column(length = 15)
    @Length(min = 2, max = 15)
    private String password;

    @OneToMany(cascade = CascadeType.REMOVE, mappedBy = "companyId",orphanRemoval = true)
    @Singular
    private List<Coupon> coupons;

}

File: Frontend/src/Components/Models/Company.ts
import { time } from "console";
import { StringLiteral } from "typescript";

export class Company {
    public id: number;
    public name: string;
    public email: string;
    public password: string;
    public coupons: any;
 
   

    constructor(id:number, name:string, email:string, password:string,coupons:any ){
        this.id = id;
        this.name = name;
        this.email = email;
        this.password = password;
        this.coupons = coupons;
       
       
    }
}